feat(auth): armazenar token JWT no MongoDB após login do usuário

Pedrohsantosg opened the pull request:
@samuelcamilocosta

Pedrohsantosg committed:
feat(auth): armazenar token JWT no MongoDB após login do usuário

samuelcamilocosta commented:
Ok, o Token agora estará armazenado, mas e ai?
Por que armazenar ele se não adicionou a comparação e busca do token ao banco?
Acabou ficando incompleto a feature

Pedrohsantosg committed:
feat(auth): adicionar verificação e comparação do token JWT armazenado no banco

- Atualizado o  para buscar o usuário pelo  e comparar o token da requisição com o token armazenado no banco de dados.
- Implementada verificação adicional para negar acesso caso o token da requisição não corresponda ao token salvo no banco, garantindo segurança adicional.
- Adicionado método  no  para permitir a invalidação do token, removendo-o do banco de dados e finalizando a sessão ativa do usuário.
- Completa a feature de gerenciamento de token JWT, permitindo controle de sessão e logout seguro.

samuelcamilocosta commented on src/controller/AuthController.ts:
```diff
@@ -51,4 +52,21 @@ export class AuthController {
         }
     }
+
+    async logout(req: Request, res: Response): Promise<Response> {
```
faltou documentar a função com TSDoc aqui

samuelcamilocosta replied:
modificador de public

samuelcamilocosta replied:
adição da rota de logout nos endpoints

Pedrohsantosg committed:
docs(auth): adicionar documentação TSDoc e ajustes de visibilidade
